include/sisl/cache/simple_cache.hpp

```diff
@@ -42,14 +43,27 @@ class SimpleCache {
             m_key_extract_cb{std::move(extract_cb)},
             m_map{num_buckets, m_key_extract_cb, std::bind(&SimpleCache< K, V >::on_hash_operation, this, _1, _2, _3)},
             m_per_value_size{per_val_size} {
-        m_record_family_id = m_evictor->register_record_family(std::move(evict_cb));
+        m_record_family_id = m_evictor->register_record_family(std::move(evict_cb), [this](const CacheRecord& record) {
+            V const value = reinterpret_cast<SingleEntryHashNode< V >*>(const_cast< CacheRecord* >(&record))->m_value;
+            K key = m_key_extract_cb(value);
+            t_evicted_keys.insert(key);
+        });
     }
 
     ~SimpleCache() { m_evictor->unregister_record_family(m_record_family_id); }
 
     bool insert(const V& value) {
         K k = m_key_extract_cb(value);
-        return m_map.insert(k, value);
+        bool ret = m_map.insert(k, value);
+        // erase evicted keys
+        if (t_evicted_keys.size() > 0) {
+            for (const auto& key : t_evicted_keys) {
+                V out_val;
```

Comment:
```
// honor insert
if (key == k) continue;
```

Reply:
The `t_evicted_keys` will never include k (the key we are trying to insert).

Reply:
Can you explain more? If a key just evicted then insert again why its not inthe map

Reply:
`t_evicted_keys` is a thread local set. Any new insert in this thread will wait until the current insert returns. 
If another thread inserts parallelly, we will remove it here which is okay IMO